### PR TITLE
[demo] - add ingress 

### PR DESCRIPTION
**Pull request by puckpuck**
Fixes: #525 

Adds capability to specify Ingress(es) for each component. This follows the same pattern used for the collector chart.

The following config will create an Ingress using the default ingress controller.  As with the collector chart, annotations are also supported.

```
components:
  frontendProxy:
    ingress:
      enabled: true
      hosts:
        - host: otel-demo.example.com
          paths:
            - path: /
              pathType: Prefix
              port: 8080
```

### CODE REVIEW

**Comment by puckpuck** [comment]
Linters are happy, this one is ready for review now

**Review by dmitryax — APPROVED** [review]
LGTM. @puckpuck please rebase

File: charts/opentelemetry-demo/templates/_helpers.tpl [hunk 1]
```diff
@@ -57,3 +57,30 @@ app.kubernetes.io/component: {{ .name}}
 {{- $mergedEnvs = concat $mergedEnvs $envOverrides }}
 {{- mustToJson $mergedEnvs }}
 {{- end }}
+
+{{/*
+Return the appropriate apiVersion for ingress.
+*/}}
+{{- define "ingress.apiVersion" -}}
+  {{- if and (.Capabilities.APIVersions.Has "networking.k8s.io/v1") (semverCompare ">= 1.19-0" .Capabilities.KubeVersion.Version) -}}
```
Comment (TylerHelmuth) [inline charts/opentelemetry-demo/templates/_helpers.tpl:65]: Is this to support old versions of k8s?  I'd rather the chart target only versions of k8s that the k8s team are still supporting.

File: charts/opentelemetry-demo/templates/_objects.tpl [hunk 2]
```diff
@@ -82,3 +83,72 @@ spec:
     {{- include "otel-demo.selectorLabels" . | nindent 4 }}
 {{- end}}
 {{- end}}
+
+{{- define "otel-demo.ingress" }}
```
Comment (TylerHelmuth) [inline charts/opentelemetry-demo/templates/_objects.tpl:87]: I think there is enough logic here to warrant a new CI scenario that tests the chart with ingress enabled.  The collector chart does the same.

File: charts/opentelemetry-demo/templates/_helpers.tpl (hunk 1 above)
Comment (puckpuck) [reply charts/opentelemetry-demo/templates/_helpers.tpl:65]: yeah, the code was copied from the Collector chart.  I'm happy to simplify this so we only support recent versions of K8s.  What would be the min version we are targeting for support? 1.23?

Comment (TylerHelmuth) [reply charts/opentelemetry-demo/templates/_helpers.tpl:65]: Dang collector chart, the maintainers of that thing should really up their game lol

I believe 1.23 is the minimum supported version ya.

File: charts/opentelemetry-demo/templates/_objects.tpl (hunk 2 above)
Comment (puckpuck) [reply charts/opentelemetry-demo/templates/_objects.tpl:87]: Added 2 new CI scenarios, and 1 example

**Review by TylerHelmuth — APPROVED** [review]

**Pull request merged**